diff --git a/makemake.sh b/makemake.sh
index d38c6fd..a49d4ab 100644
--- a/makemake.sh
+++ b/makemake.sh
@@ -370,7 +370,7 @@ fi
 # stack trace of the issue. If one wishes, one can run 'strip -g Mlucas' to remove the debugging symbols:
 cat <<EOF >Makefile
 CC ?= gcc
-CFLAGS = -fdiagnostics-color -Wall -g -O3 -flto # =auto
+CFLAGS = -fdiagnostics-color -Wall -g -O3 -flto -Wno-error=incompatible-pointer-types # =auto
 CPPFLAGS ?= -I/usr/local/include -I/opt/homebrew/include
 LDFLAGS ?= -L/opt/homebrew/lib
 LDLIBS = ${LD_ARGS[@]} # -static
